fix: update and add timeouts (#1434)
saucelabs/sauce-docs: docs/dev/test-configuration-options.md

<p><small>| OPTIONAL | BOOLEAN | <span className="sauceDBlue">Virtual and Real Devices</span> |</small></p>

> Prevents the device to be reset before the session startup if set to `true`. This means that the application under test is not going to be terminated neither its data cleaned. This capability behaves differently across virtual and real devices.

Reviewer: ```suggestion
If set to `true` it prevents the device from resetting before the session startup. This means the application under test will not be terminated or its data cleaned. This capability behaves differently across virtual and real devices.
```

**For Real Devices:**

Set `noReset` to `true` to keep a device allocated to you during the device cleaning process, as described under [`cacheId`](#`cacheId`), allowing you to continue testing on the same device. Default value is `false`. To use `noReset`, you must pair it with `cacheId`.

**For Virtual Devices:**

> This capability will have no effect on Sauce Labs virtual devices, it will only work on local Android emulators / iOS simulators. For local executions you will likely only have one device available, in Sauce Labs you will have a pool of devices available, depending on your concurrency. Each session will start a new clean session which will make this capability redundant.

Reviewer: ```suggestion
This capability will have no effect on Sauce Labs virtual devices, it will only work on local Android Emulators/iOS Simulators. For local executions, you will likely only have one device available, in Sauce Labs you will have a pool of devices available depending on your concurrency. Each session will start a new clean session which will make this capability redundant.
```

:::

```java
MutableCapabilities capabilities = new MutableCapabilities();
capabilities.setCapability("appium:autoGrantPermissions", true);
```

## Mobile Appium Timeout Capabilities
As with Selenium Tests, Appium also supports different types of timeouts like:
- [Implicit Wait Timeout](https://appium.io/docs/en/commands/session/timeouts/implicit-wait/#set-implicit-wait-timeout): Set the amount of time the driver should wait when searching for elements
- [Script Timeouts](https://appium.io/docs/en/commands/session/timeouts/async-script/index.html): Sets the amount of time, in milliseconds, that asynchronous scripts executed by [execute async](https://appium.io/docs/en/commands/web/execute-async/index.html) are permitted to run before they are aborted (Web context only)

These timeouts can be controlled by the driver during the test session. There are timeouts that can be set as a capability for when you start the driver and can be driver specific.

---

### `appium:newCommandTimeout`

<p><small>| DURATION | INTEGER | <span className="sauceDBlue">Virtual and Real Devices</span> | <span className="sauceDBlue">Android and iOS</span> |</small></p>

> Sets the amount of time in seconds how long the driver should wait for a new command from the client before assuming the clients has stopped sending requests. If there is no response during this time, the next executed command on the Virtual/Real Device will time out. The default value is 60 seconds while the maximum allowed value is not limited for Virtual Devices and is limited to 90 seconds for Real Devices.

Reviewer: ```suggestion
Specifies the amount of time in seconds, in which the driver waits for a new command from the client before assuming the client has stopped sending requests. If there is no response during this time, the next executed command on the Virtual/Real Device will time out. The default value is 60 seconds while the maximum allowed value is not limited for Virtual Devices and is limited to 90 seconds for Real Devices.
```
